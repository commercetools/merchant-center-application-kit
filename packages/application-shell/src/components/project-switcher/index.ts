export { default } from './project-switcher';
