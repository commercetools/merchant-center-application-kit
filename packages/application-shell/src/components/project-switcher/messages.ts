import { defineMessages } from 'react-intl';

export default defineMessages({
  projects: {
    id: 'ProjectSwitcher.projects',
    description: 'The label for project dropdown switcher',
    defaultMessage: 'Projects',
  },
  searchPlaceholder: {
    id: 'ProjectSwitcher.searchPlaceholder',
    description: 'Projects filter search input placeholder',
    defaultMessage: 'Search for a project',
  },
  noResults: {
    id: 'ProjectSwitcher.noResults',
    description: 'Projects filter no results text',
    defaultMessage: 'Sorry, but there are no projects that match your search.',
  },
  suspended: {
    id: 'ProjectSwitcher.suspended',
    description: 'Label for a suspended project',
    defaultMessage: 'Suspended',
  },
  expired: {
    id: 'ProjectSwitcher.expired',
    description: 'Label for a suspended project',
    defaultMessage: 'Expired',
  },
});
