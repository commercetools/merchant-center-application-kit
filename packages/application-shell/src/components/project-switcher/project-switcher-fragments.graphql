fragment projectFragment on Project {
  name
  key
  suspension {
    isActive
  }
  expiry {
    isActive
  }
}
