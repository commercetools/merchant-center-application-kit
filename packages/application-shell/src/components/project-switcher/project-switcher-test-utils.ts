import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import ProjectsQuery from './project-switcher.mc.graphql';

type CreateGraphqlResponseForProjectsQueryOptions = {
  numberOfProjects?: number;
  getIsSuspended?: (key: string) => boolean;
  getIsExpired?: (key: string) => boolean;
};

const falsy = () => false;

export const createGraphqlResponseForProjectsQuery = ({
  numberOfProjects = 4,
  getIsSuspended = falsy,
  getIsExpired = falsy,
}: CreateGraphqlResponseForProjectsQueryOptions = {}) => ({
  request: {
    query: ProjectsQuery,
    context: { target: GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND },
  },
  result: {
    data: {
      loading: false,
      user: {
        __typename: 'User',
        id: 'user-id',
        projects: {
          __typename: 'ProjectQueryResult',
          results: Array.from({ length: numberOfProjects }).map((_, index) => {
            const key = `key-${index}`;
            const name = `Name ${index}`;
            return {
              __typename: 'Project',
              key,
              name,
              suspension: {
                __typename: 'ProjectSuspension',
                isActive: getIsSuspended(key),
              },
              expiry: {
                __typename: 'ProjectExpiry',
                isActive: getIsExpired(key),
              },
            };
          }),
        },
      },
    },
  },
});
