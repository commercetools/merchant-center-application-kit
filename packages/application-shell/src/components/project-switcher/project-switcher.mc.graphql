#import "./project-switcher-fragments.mc.graphql"

query ProjectsQuery {
  user: me {
    id
    projects {
      results {
        ...projectFragment
      }
    }
  }
}
