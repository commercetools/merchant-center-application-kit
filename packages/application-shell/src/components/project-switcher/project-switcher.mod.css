.react-select-wrapper {
  width: 225px;
}

.option-container {
  word-wrap: break-word;
}

.single-value-container {
  flex: 1;
  align-items: center;
  display: flex;
}

.dropdown-container {
  display: flex;
  align-items: center;
  height: 100%;
}

.ellipsis {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.project-name {
  color: var(--color-accent);
}

.value-container {
  display: flex;
  flex: 1;
  align-items: center;
}

.value-wrapper {
  flex: 1;
}

.project-counter {
  width: 22px;
  height: 22px;
  border-radius: 22px;
  padding: 3px;
  background: var(--color-accent-40);
  color: var(--color-surface);
  display: inline-block;
  font-size: 0.9rem;
  text-align: center;
}

.item-text-main {
  color: var(--color-accent);
}

.item-text-disabled {
  color: var(--color-neutral);
}

.item-text-main.item-text-disabled {
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
}

.item-text-small {
  font-size: 11px;
}

.red {
  color: var(--color-error);
}

.disabled-icon-container {
  font-size: 1.5rem;
  display: flex;
}
