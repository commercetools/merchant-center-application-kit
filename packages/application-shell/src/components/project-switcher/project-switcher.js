import React from 'react';
import PropTypes from 'prop-types';
import { defaultMemoize } from 'reselect';
import { graphql } from 'react-apollo';
import { FormattedMessage, injectIntl } from 'react-intl';
import flowRight from 'lodash/flowRight';
import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import classnames from 'classnames';
import { SelectInput, ErrorIcon } from '@commercetools-frontend/ui-kit';
import styles from './project-switcher.mod.css';
import ProjectsQuery from './project-switcher.graphql';
import messages from './messages';

export const ValueContainer = props => (
  <div className={styles['value-container']}>
    <div className={styles['value-wrapper']}>
      <SelectInput.ValueContainer {...props}>
        {props.children}
      </SelectInput.ValueContainer>
    </div>
    <span className={styles['project-counter']}>{props.projectCount}</span>
  </div>
);

ValueContainer.propTypes = {
  children: PropTypes.node.isRequired,
  projectCount: PropTypes.number.isRequired,
};

ValueContainer.displayName = 'ValueContainer';

export const Option = props => (
  <SelectInput.Option {...props}>
    <div className={styles['option-container']}>
      <div
        className={classnames(styles['item-text-main'], {
          [styles['item-text-disabled']]:
            (props.data.suspension && props.data.suspension.isActive) ||
            (props.data.expiry && props.data.expiry.isActive),
        })}
      >
        {props.data.name}
        {((props.data.suspension && props.data.suspension.isActive) ||
          (props.data.expiry && props.data.expiry.isActive)) && (
          <span className={styles['disabled-icon-container']}>
            <ErrorIcon size="medium" />
          </span>
        )}
      </div>
      <div
        className={classnames(styles['item-text-small'], {
          [styles['item-text-disabled']]:
            (props.data.suspension && props.data.suspension.isActive) ||
            (props.data.expiry && props.data.expiry.isActive),
        })}
      >
        {props.data.key}
      </div>
      {props.data.suspension && props.data.suspension.isActive && (
        <div className={classnames(styles.red, styles['item-text-small'])}>
          <FormattedMessage {...messages.suspended} />
        </div>
      )}
      {props.data.expiry && props.data.expiry.isActive && (
        <div className={classnames(styles.red, styles['item-text-small'])}>
          <FormattedMessage {...messages.expired} />
        </div>
      )}
    </div>
  </SelectInput.Option>
);

Option.propTypes = {
  data: PropTypes.shape({
    key: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    expiry: PropTypes.shape({
      isActive: PropTypes.bool,
    }).isRequired,
    suspension: PropTypes.shape({
      isActive: PropTypes.bool,
    }).isRequired,
  }).isRequired,
};

Option.displayName = 'Option';

const mapProjectsToOptions = defaultMemoize(projects =>
  projects.map(project => ({
    key: project.key,
    name: project.name,
    label: project.name,
    value: project.key,
    suspension: project.suspension,
    expiry: project.expiry,
  }))
);

export class ProjectSwitcher extends React.PureComponent {
  static displayName = 'ProjectSwitcher';

  static propTypes = {
    projectKey: PropTypes.string.isRequired,
    total: PropTypes.number.isRequired,
    // injectIntl
    intl: PropTypes.shape({
      formatMessage: PropTypes.func.isRequired,
    }).isRequired,
    // graphql
    projectsQuery: PropTypes.shape({
      loading: PropTypes.bool.isRequired,
      error: PropTypes.shape({
        message: PropTypes.string.isRequired,
      }),
      user: PropTypes.shape({
        projects: PropTypes.shape({
          results: PropTypes.arrayOf(
            PropTypes.shape({
              key: PropTypes.string.isRequired,
              name: PropTypes.string.isRequired,
              suspension: PropTypes.shape({
                isActive: PropTypes.bool.isRequired,
              }),
              expiry: PropTypes.shape({
                isActive: PropTypes.bool.isRequired,
              }),
            })
          ),
        }),
      }),
    }),
  };

  redirectTo = targetUrl => window.location.replace(targetUrl);

  getProjectName = key => {
    const selectedProject = this.props.projectsQuery.user.projects.results.find(
      project => project.key === key
    );
    return selectedProject && selectedProject.name;
  };

  handleSelection = ({ target: { value: selectedProjectKey } }) => {
    if (selectedProjectKey !== this.props.projectKey)
      // We simply redirect to a "new" browser page, instead of using the
      // history router. This will simplify a lot of things and avoid possible
      // problems like e.g. resetting the store/state.
      this.redirectTo(`/${selectedProjectKey}`);
  };

  render() {
    if (this.props.projectsQuery.loading) return null;

    return (
      <div
        className={styles['react-select-wrapper']}
        data-track-component="ProjectSwitch"
        data-track-event="click"
      >
        <SelectInput
          value={this.props.projectKey}
          name="project-switcher"
          onChange={this.handleSelection}
          options={
            this.props.projectsQuery.user &&
            mapProjectsToOptions(this.props.projectsQuery.user.projects.results)
          }
          components={{
            Option,
            ValueContainer: props => (
              <ValueContainer
                {...props}
                projectCount={
                  this.props.projectsQuery.user.projects.results.length
                }
              />
            ),
          }}
          isClearable={false}
          backspaceRemovesValue={false}
          placeholder={this.props.intl.formatMessage(
            messages.searchPlaceholder
          )}
          noOptionsMessage={() =>
            this.props.intl.formatMessage(messages.noResults)
          }
        />
      </div>
    );
  }
}

export default flowRight(
  graphql(ProjectsQuery, {
    name: 'projectsQuery',
    options: () => ({
      variables: {
        target: GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND,
      },
    }),
  }),
  injectIntl
)(ProjectSwitcher);
