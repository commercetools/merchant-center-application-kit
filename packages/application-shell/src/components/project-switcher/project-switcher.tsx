import React from 'react';
import memoize from 'memoize-one';
import { useQuery } from 'react-apollo';
import { FormattedMessage, useIntl } from 'react-intl';
import { OptionProps, ValueContainerProps } from 'react-select';
import { css } from '@emotion/core';
import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import { reportErrorToSentry } from '@commercetools-frontend/sentry';
import SelectInput from '@commercetools-uikit/select-input';
import { ErrorIcon } from '@commercetools-uikit/icons';
import { customProperties } from '@commercetools-uikit/design-system';
import { location } from '../../utils/location';
import {
  TProject,
  TFetchUserProjectsQuery,
  TFetchUserProjectsQueryVariables,
} from '../../types/generated/mc';
import ProjectsQuery from './project-switcher.mc.graphql';
import messages from './messages';

type Props = {
  projectKey?: string;
};
type OptionType = Pick<TProject, 'key' | 'name' | 'suspension' | 'expiry'> & {
  label: string;
};
type CustomValueContainerProps = ValueContainerProps<OptionType> & {
  projectCount: number;
};

export const ProjectSwitcherValueContainer = ({
  projectCount,
  ...restProps
}: CustomValueContainerProps) => (
  <div
    css={css`
      display: flex;
      flex: 1;
      align-items: center;
    `}
  >
    <div
      css={css`
        flex: 1;
      `}
    >
      <SelectInput.ValueContainer {...restProps}>
        {restProps.children}
      </SelectInput.ValueContainer>
    </div>
    <span
      css={css`
        width: 22px;
        height: 22px;
        border-radius: 100%;
        background: ${customProperties.colorAccent40};
        color: ${customProperties.colorSurface};
        font-size: 0.9rem;
        display: flex;
        justify-content: center;
        align-items: center;
      `}
    >
      {projectCount}
    </span>
  </div>
);

export const ProjectSwitcherOption = (props: OptionProps<OptionType>) => (
  <SelectInput.Option {...props}>
    <div
      css={css`
        word-wrap: break-word;
      `}
    >
      <div
        css={css`
          color: ${props.isDisabled
            ? customProperties.colorNeutral
            : customProperties.colorAccent};
        `}
      >
        {props.data.name}
        {props.isDisabled && (
          <span
            css={css`
              font-size: 1.5rem;
              display: flex;
            `}
          >
            <ErrorIcon size="medium" />
          </span>
        )}
      </div>
      <div
        css={css`
          font-size: 11px;
          color: ${props.isDisabled
            ? customProperties.colorNeutral
            : 'inherit'};
        `}
      >
        {props.data.key}
      </div>
      {props.data.suspension && props.data.suspension.isActive && (
        <div
          css={css`
            font-size: 11px;
            color: ${customProperties.colorError};
          `}
        >
          <FormattedMessage {...messages.suspended} />
        </div>
      )}
      {props.data.expiry && props.data.expiry.isActive && (
        <div
          css={css`
            font-size: 11px;
            color: ${customProperties.colorError};
          `}
        >
          <FormattedMessage {...messages.expired} />
        </div>
      )}
    </div>
  </SelectInput.Option>
);

const mapProjectsToOptions = memoize((projects) =>
  projects.map((project: TProject) => ({
    key: project.key,
    name: project.name,
    label: project.name,
    value: project.key,
    suspension: project.suspension,
    expiry: project.expiry,
  }))
);

const redirectTo = (targetUrl: string) => location.replace(targetUrl);

const ProjectSwitcher = (props: Props) => {
  const intl = useIntl();
  const { loading, data } = useQuery<
    TFetchUserProjectsQuery,
    TFetchUserProjectsQueryVariables
  >(ProjectsQuery, {
    onError: reportErrorToSentry,
    variables: {
      target: GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND,
    },
  });

  if (loading) return null;

  return (
    <div
      css={css`
        width: 225px;
      `}
      data-track-component="ProjectSwitch"
      data-track-event="click"
    >
      <SelectInput<OptionType>
        value={props.projectKey || ''}
        name="project-switcher"
        aria-labelledby="project-switcher"
        onChange={(event) => {
          const selectedProjectKey = event.target.value;
          if (selectedProjectKey !== props.projectKey)
            // We simply redirect to a "new" browser page, instead of using the
            // history router. This will simplify a lot of things and avoid possible
            // problems like e.g. resetting the store/state.
            redirectTo(`/${selectedProjectKey}`);
        }}
        options={
          data && data.user && mapProjectsToOptions(data.user.projects.results)
        }
        isOptionDisabled={(option) =>
          option.suspension.isActive || option.expiry.isActive
        }
        components={{
          // eslint-disable-next-line react/display-name
          Option: (optionsProps) => <ProjectSwitcherOption {...optionsProps} />,
          // eslint-disable-next-line react/display-name
          ValueContainer: (valueContainerProps) => (
            <ProjectSwitcherValueContainer
              {...valueContainerProps}
              projectCount={
                (data && data.user && data.user.projects.results.length) || 0
              }
            />
          ),
        }}
        isClearable={false}
        backspaceRemovesValue={false}
        placeholder={intl.formatMessage(messages.searchPlaceholder)}
        noOptionsMessage={() => intl.formatMessage(messages.noResults)}
      />
    </div>
  );
};
ProjectSwitcher.displayName = 'ProjectSwitcher';

export default ProjectSwitcher;
