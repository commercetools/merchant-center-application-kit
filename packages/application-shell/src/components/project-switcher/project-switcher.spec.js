import { shallow } from 'enzyme';
import React from 'react';
import { SelectInput } from '@commercetools-frontend/ui-kit';
import { ProjectSwitcher, Option, ValueContainer } from './project-switcher';

const createProjectsQuery = custom => ({
  loading: false,
  user: {
    projects: {
      results: [
        {
          key: 'key4',
          name: 'name4',
          suspension: { isActive: false },
          expiry: { isActive: false },
        },
        {
          key: 'key2',
          name: 'name2',
          suspension: { isActive: false },
          expiry: { isActive: false },
        },
        {
          key: 'key1',
          name: 'name1',
          suspension: { isActive: false },
          expiry: { isActive: false },
        },
        {
          key: 'key3',
          name: 'name3',
          suspension: { isActive: false },
          expiry: { isActive: false },
        },
      ],
    },
  },

  ...custom,
});

const createTestProps = props => ({
  projectKey: 'key1',
  total: 4,
  projectsQuery: createProjectsQuery(),
  intl: { formatMessage: jest.fn() },
  ...props,
});

describe('rendering', () => {
  let wrapper;
  let props;

  describe('when loading', () => {
    beforeEach(() => {
      props = createTestProps({
        projectsQuery: createProjectsQuery({
          loading: true,
        }),
      });
      wrapper = shallow(<ProjectSwitcher {...props} />);
    });

    it('should not render a `<SelectInput>`', () => {
      expect(wrapper).not.toRender(SelectInput);
    });
  });

  describe('without `user` in `projectsQuery`', () => {
    beforeEach(() => {
      props = createTestProps({
        projectsQuery: createProjectsQuery({
          user: null,
        }),
      });
      wrapper = shallow(<ProjectSwitcher {...props} />);
    });

    it('should render a `<SelectInput>`', () => {
      expect(wrapper).toRenderElementTimes(SelectInput, 1);
    });

    it('should pass empty `options` to `<SelectInput>`', () => {
      expect(wrapper.find(SelectInput)).toHaveProp('options', null);
    });
  });

  describe('when loaded', () => {
    let valueContainer;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<ProjectSwitcher {...props} />);
      valueContainer = shallow(
        <ValueContainer
          projectCount={props.projectsQuery.user.projects.results.length}
        >
          test project
        </ValueContainer>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a `<SelectInput>`', () => {
      expect(wrapper).toRenderElementTimes(SelectInput, 1);
    });

    it('should render a `<SelectInput>` with the value set to the current project', () => {
      expect(wrapper.find(SelectInput).prop('value')).toBe(props.projectKey);
    });

    it('should render the number of projects in the selector', () => {
      expect(valueContainer).toIncludeText('4');
    });

    describe('dropdown option item', () => {
      describe('project is not expired or suspended', () => {
        let optionWrapper;
        beforeEach(() => {
          const data = {
            name: 'Project A',
            key: 'project-a',
            expiry: { isActive: false },
            suspension: { isActive: false },
          };

          optionWrapper = shallow(<Option data={data} />);
        });
        it('should render item name', () => {
          expect(optionWrapper.find('.item-text-main')).toIncludeText(
            'Project A'
          );
        });
        it('should render item key', () => {
          expect(optionWrapper.find('.item-text-small').at(0)).toIncludeText(
            'project-a'
          );
        });
        it('should not render suspended message', () => {
          expect(optionWrapper).not.toRender('FormattedMessage');
        });
        it('should not render expired message', () => {
          expect(optionWrapper).not.toRender('FormattedMessage');
        });
      });
      describe('project is has suspension', () => {
        let optionWrapper;
        beforeEach(() => {
          const data = {
            name: 'Project A',
            key: 'project-a',
            expiry: { isActive: false },
            suspension: { isActive: true },
          };

          optionWrapper = shallow(<Option data={data} />);
        });
        it('should render item name', () => {
          expect(
            optionWrapper.find('.item-text-main.item-text-disabled')
          ).toIncludeText('Project A');
        });
        it('should render `ErrorIcon`', () => {
          expect(optionWrapper).toRender('ErrorIcon');
        });
        it('should render item key', () => {
          expect(
            optionWrapper.find('.item-text-small.item-text-disabled')
          ).toIncludeText('project-a');
        });
        it('should render suspended message', () => {
          expect(optionWrapper).toRender({ id: 'ProjectSwitcher.suspended' });
        });
        it('should not render expired message', () => {
          expect(optionWrapper).not.toRender({ id: 'ProjectSwitcher.expired' });
        });
      });
      describe('project is expired', () => {
        let optionWrapper;
        beforeEach(() => {
          const data = {
            name: 'Project A',
            key: 'project-a',
            expiry: { isActive: true },
            suspension: {
              isActive: false,
            },
          };

          optionWrapper = shallow(<Option data={data} />);
        });
        it('should render item name', () => {
          expect(
            optionWrapper.find('.item-text-main.item-text-disabled')
          ).toIncludeText('Project A');
        });
        it('should render `ErrorIcon`', () => {
          expect(optionWrapper).toRender('ErrorIcon');
        });
        it('should render item key', () => {
          expect(
            optionWrapper.find('.item-text-small.item-text-disabled')
          ).toIncludeText('project-a');
        });
        it('should render expired message', () => {
          expect(optionWrapper).toRender({ id: 'ProjectSwitcher.expired' });
        });
        it('should not render suspended message', () => {
          expect(optionWrapper).not.toRender({
            id: 'ProjectSwitcher.suspended',
          });
        });
      });
    });
  });
});

describe('callbacks', () => {
  describe('selecting a project from the list', () => {
    let wrapper;
    let props;

    describe('when the user selects a different project from the current one', () => {
      let replace;
      beforeEach(() => {
        replace = jest.fn();
        delete window.location;
        window.location = { replace };
        props = createTestProps();
        wrapper = shallow(<ProjectSwitcher {...props} />);
        wrapper
          .find(SelectInput)
          .simulate('change', { target: { value: 'key2' } });
      });
      it('should redirect to the new project url', () => {
        expect(replace).toHaveBeenCalledTimes(1);
        expect(replace).toHaveBeenLastCalledWith('/key2');
      });
    });

    describe('when the user selects same project as the current one', () => {
      let replace;
      beforeEach(() => {
        replace = jest.fn();
        delete window.location;
        window.location = { replace };
        props = createTestProps({ projectKey: 'key1' });
        wrapper = shallow(<ProjectSwitcher {...props} />);
        wrapper
          .find(SelectInput)
          .simulate('change', { target: { value: 'key1' } });
      });
      it('should not redirect to the new project', () => {
        expect(replace).toHaveBeenCalledTimes(0);
      });
    });
  });
});
