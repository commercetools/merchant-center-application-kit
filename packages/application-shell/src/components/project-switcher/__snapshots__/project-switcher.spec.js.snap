// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering when loaded should match snapshot 1`] = `
<div
  className="react-select-wrapper"
  data-track-component="ProjectSwitch"
  data-track-event="click"
>
  <InjectIntl(SelectInput)
    backspaceRemovesValue={false}
    components={
      Object {
        "Option": [Function],
        "ValueContainer": [Function],
      }
    }
    isClearable={false}
    name="project-switcher"
    noOptionsMessage={[Function]}
    onChange={[Function]}
    options={
      Array [
        Object {
          "expiry": Object {
            "isActive": false,
          },
          "key": "key4",
          "label": "name4",
          "name": "name4",
          "suspension": Object {
            "isActive": false,
          },
          "value": "key4",
        },
        Object {
          "expiry": Object {
            "isActive": false,
          },
          "key": "key2",
          "label": "name2",
          "name": "name2",
          "suspension": Object {
            "isActive": false,
          },
          "value": "key2",
        },
        Object {
          "expiry": Object {
            "isActive": false,
          },
          "key": "key1",
          "label": "name1",
          "name": "name1",
          "suspension": Object {
            "isActive": false,
          },
          "value": "key1",
        },
        Object {
          "expiry": Object {
            "isActive": false,
          },
          "key": "key3",
          "label": "name3",
          "name": "name3",
          "suspension": Object {
            "isActive": false,
          },
          "value": "key3",
        },
      ]
    }
    value="key1"
  />
</div>
`;
